Mark Token Pruning for Sparse Vector as GA

markjhoy opened the pull request:
A docs change to remove the [preview] labels for the token pruning configuration items for the Sparse Vector query as we get ready to put these into GA.

Commit:
remove [preview] labels sparse vec / token pruning

Commit:
Update docs/changelog/128854.yaml

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

elasticsearchmachine commented:
Hi @markjhoy, I've created a changelog YAML for you. *Note* that since this PR is labelled `release highlight`, you need to update the changelog YAML to fill out the extended information sections.

elasticsearchmachine commented:
Pinging @elastic/search-relevance (Team:Search - Relevance)

Commit:
update changelog

Samiul-TheSoccerFan reviewed (approved):
LGTM

kderusso commented on docs/changelog/128854.yaml:
```diff
@@ -0,0 +1,11 @@
+pr: 128854
+summary: Mark token pruning for sparse vector as GA
+area: Relevance
```
Should this be `ml`?

kderusso commented on docs/changelog/128854.yaml:
```diff
@@ -0,0 +1,11 @@
+pr: 128854
+summary: Mark token pruning for sparse vector as GA
+area: Relevance
+type: docs
```
I think we want to change this to `feature` not `docs` as it's a release highlight. See https://github.com/elastic/elasticsearch/pull/125302 as an example

kderusso reviewed (approved):
Looks good, just need to clean up a couple issues of release notes and fix the build. 

Could you please also open a PR to mark this as stable in the spec? See: https://github.com/elastic/elasticsearch/pull/125302/files

Thanks!

Commit:
set changelog to feature/ml

markjhoy commented:
> Could you please also open a PR to mark this as stable in the spec? See: https://github.com/elastic/elasticsearch/pull/125302/files

@kderusso - what spec are you referring to?

Commit:
set proper area

markjhoy commented:
NOTE: do not merge this yet - wait at least until https://github.com/elastic/elasticsearch/pull/126739/ is merged, and we consult appropriate stakeholders

leemthompo commented on docs/reference/query-languages/query-dsl/query-dsl-sparse-vector-query.md:
```diff
@@ -63,21 +63,21 @@ GET _search
 :   (Optional, dictionary) A dictionary of token-weight pairs representing the precomputed query vector to search. Searching using this query vector will bypass additional inference. Only one of `inference_id` and `query_vector` is allowed.
 
 `prune`
```
```suggestion
`prune` {applies_to}`stack: preview 9.0, ga 9.1`
```

leemthompo reviewed (changes requested):
Will need to update the `applies_to` versioning metadata before we can merge, thanks :)

Commit:
add applies_to labels

Commit:
Merge branch 'main' into markjhoy/mark_token_pruning_ga

leemthompo reviewed (approved):
Thanks Mark!

leemthompo commented:
> Could you please also open a PR to mark this as stable in the spec? See: https://github.com/elastic/elasticsearch/pull/125302/files
@kderusso - what spec are you referring to?

I presume this means updating https://github.com/elastic/elasticsearch-specification/blob/main/specification/_types/query_dsl/SparseVectorQuery.ts#L60-L80

markjhoy commented:
@leemthompo - there should already be a PR for the specification here: https://github.com/elastic/elasticsearch-specification/pull/4454

Commit:
add clarification for token pruning behaviour

markjhoy commented:
@kderusso and @leemthompo - a quick update to the docs here - I added in the updated detailed description of pruning behaviour (what gets pruned and what does not) from the docs we added for the sparse vector index options (copy/paste from that PR to here and these docs).

kderusso reviewed (commented):
Good changes, thanks for adding them! One note on versioning

kderusso commented on docs/reference/query-languages/query-dsl/query-dsl-sparse-vector-query.md:
```diff
@@ -62,28 +62,36 @@ GET _search
 
-`prune`
-:   (Optional, boolean) [preview] Whether to perform pruning, omitting the non-significant tokens from the query to improve query performance. If `prune` is true but the `pruning_config` is not specified, pruning will occur but default values will be used. Default: false.
+`prune` {applies_to}`stack: preview 9.0, ga 9.1`
```
But this is net new for 9.1.

kderusso replied:
But then it would be when it was released which was prior to 9.0.

leemthompo replied:
>  Or since these docs are 9.0+ only? This is really confusing as a docs structure, sorry :(

Yup the the 9.x docs don't talk about 8.x and below in the applies_to metadata. So yeah it doesn't mean "this feature was added in 9.0 in tech preview", it just means "This feature was tech preview in 9.0".

The actual output on the page will have more helpful tooltips and stuff in the future, but yeah the raw metadata is a bit ambiguous. It's also hard to get used to the "[cumulative docs](https://elastic.github.io/docs-builder/contribute/?v=426B04AA#cumulative-docs)" paradigm but we need it because this page applies to both 9.0 and 9.1 (and beyond as time goes on), so in order to specify when something went GA in 9.1, we need to also be able too specify it was preview in 9.0. Because there's no 9.0 page to switch back to.
